bot/utils/channel.py

```diff
@@ -15,6 +16,15 @@ def is_help_channel(channel: discord.TextChannel) -> bool:
     return any(is_in_category(channel, category) for category in categories)
 
 
+def is_mod_channel(channel: discord.TextChannel) -> bool:
+    """Return True if `channel` is one of the moderation channels or in one of the moderation categories."""
+    log.trace(f"Checking if #{channel} is a mod channel.")
+    categories = (Categories.modmail, Categories.logs)
+
+    return (channel.id in constants.MODERATION_CHANNELS
+            or any(is_in_category(channel, category) for category in categories))
```

Comment:
Prevailing style for a multi-line expression would be:
```python
return (
    a
    or b
)
```

How would you feel about handling each case separately and logging the decision made? For example:
```python
def is_mod_channel(channel: discord.TextChannel) -> bool:
    """True if `channel` is considered a mod channel."""
    if channel.id in constants.MODERATION_CHANNELS:
        log.trace(f"Channel #{channel} is a configured mod channel")
        return True

    mod_categories = Categories.modmail, Categories.logs
    if any(is_in_category(channel, category) for category in mod_categories):
        log.trace(f"Channel #{channel} is in a configured mod category")
        return True

    log.trace(f"Channel #{channel} is not a mod channel")
    return False
```

Looking at one of the callers, the infraction scheduler, it branches based on the return value just so that it can log the decision that was made. One of the clauses could be removed if we just log here. But it's just a thought, I'm not demanding this.

Finally, I wonder if the mod categories should be defined in the config like the mod channels, and then accessed here as e.g. `constants.MODERATION_CATEGORIES`. I don't necessarily think that's the best way to do it, but it'd be more consistent with the way the channel IDs are handled.

Reply:
Ok, I'll fix the expression.
Yeah, I think adding the constant is a good idea, it will indicate why those categories are needed in the config file.

As for logging, I think it largely depends on the level of granularity the core developers want for it. Handling it outside of the function allows you to add relevant context. I'm fine with whatever you think is better.